### PR TITLE
Add mountable autodoc and autodoc couch (with EgorBelphegor additions)

### PR DESCRIPTION
**Pull request by leoCottret**
#### Summary
SUMMARY: Content "Add mountable autodoc and autodoc couch"

#### Purpose of change
There is already an autoclave to sterilize CBMs, but you still had to transport your autodoc and your autodoc couch as items, and installing/removing them each time you need an autodoc.

#### Describe the solution
Add the mountable autodoc/autodoc couch items to the game.
They can do the same things as their furniture counterpart (cf tests).
They don't use power, same as their furniture counterpart.

Edit: Since @EgorBelphegor proposed an other [PR](https://github.com/cataclysmbnteam/Cataclysm-BN/pull/2140) for this feature, I merged a part of it in this one.

The most notable changes are:
- you can now operate while the vehicle is moving (even in an helicopter, and going over bridges is supported too)
- the code is a bit cleaner, I especially liked the usage of a function to check if a flag is on a furniture or a vehicle part, this could be useful in the future. I hesitated to add it when I was developping the PR, but seeing the result it is indeed much better.
- hostile nursebots are able to use autodoc vehicles too. I didn't bother to add it because the chance of this happening seemed very slim (they can only remove ~2 bionics, and they can't reload their anesthetic after that), but it's better if they can do it noneless.

So even if @EgorBelphegor probably didn't see this PR, it was still very useful, thanks.

#### Testing
- install CBM
- uninstall CBM
- splint broken limb (can splint arms and legs of people with normal sized body, same as the furniture version)
- treat wounds (can stop bleeding and other almost hidden features, same as the furniture version)
- check radiation level
- do everything above on a follower
- convert a broken cyborg (failure and success)
- repeat every actions above, but on a normal autodoc
- craft a mountable autodoc/autodoc couch

Edit: I retested everything above since the integration of @EgorBelphegor's PR

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -5101,6 +5103,21 @@ void game::save_cyborg( item *cyborg, const tripoint &couch_pos, player &install
 
 }
 
+void game::delete_cyborg_item( const tripoint &couch_pos, item *cyborg )
```
Comment (Coolthulhu) [inline src/game.cpp:5106]: I don't think it "deserves" to be a `game` function. Should just take `map &` as an argument and be a `static` function, defined right above `save_cyborg`.

File: data/json/items/vehicle/utilities.json [hunk 2]
```diff
@@ -156,6 +156,32 @@
...
+    "type": "GENERIC",
+    "id": "vh_autodoc",
+    "name": { "str": "mountable autodoc" },
+    "description": "This autodoc has been rigged to run off a vehicle power grid.",
+    "symbol": "&",
+    "color": "light_cyan",
+    "price": 120000,
+    "price_postapoc": 7500,
+    "weight": "34500 g",
+    "volume": "76 L",
```
Comment (Night-Pryanik) [inline data/json/items/vehicle/utilities.json:169]: It's up to you to decide, of course, but originally autodoc was meant to weight a couple of hundreds of kilograms. And it's definitely should not fit in RL large hiking rucksack (80 L).

File: data/json/items/vehicle/utilities.json [hunk 3]
```diff
@@ -156,6 +156,32 @@
...
+  {
+    "type": "GENERIC",
+    "id": "vh_autodoc",
+    "name": { "str": "mountable autodoc" },
+    "description": "This autodoc has been rigged to run off a vehicle power grid.",
+    "symbol": "&",
+    "color": "light_cyan",
+    "price": 120000,
+    "price_postapoc": 7500,
+    "weight": "34500 g",
```
[the remark this answers is not in the document]
Comment (leoCottret) [reply data/json/items/vehicle/utilities.json]: Yes good point, I changed the weight and volume to their furniture counterpart.

File: src/map.cpp [hunk 4]
```diff
@@ -8686,6 +8698,34 @@ std::list<tripoint> map::find_furnitures_with_flag_in_radius( const tripoint &ce
...
+        } else {
+            const tripoint newloc( loc + tripoint( 0, 0, dz ) );
+            if( has_flag_furn_or_vpart( flag, newloc ) ) {
+                locs.push_back( newloc );
+            }
+        }
+    }
+
+    return locs;
+}
```
Comment (Coolthulhu) [inline src/map.cpp:8727]: I specifically tested the up/down thing and it doesn't seem to work.
I parked a vehicle such that autodoc couch and autodoc are on different z-levels and it failed to find the couch.

Not sure if it's important to have it working. But it should at least have a comment saying that it doesn't work if x is happening. Or be turned into a TODO comment with the not-working section removed.

Comment (leoCottret) [reply src/map.cpp:8727]: I will add a comment to clarify it, but this code isn't supposed to allow interacting with the autodoc through Z levels.

This adds support for moving the vehicle through Z levels after the operation is started.
There's a check each turn to know if the player is still on the couch. 
If he's not, that means he has moved during the operation, so he should be hurt and the operation fails.

My implementation assumed that we didn't move the vehicle, so I only checked in an horizontal radius around the autodoc.
This code checks for a couch in a "3D" radius around the autodoc instead (excluding just above/below).

**Comment by Coolthulhu** [comment]
For some weird reason, autodoc couch furniture is in its own tab in construction. Nothing else is in it. Not caused here, just a note.

**Comment by chaosvolt** [comment]
> For some weird reason, autodoc couch furniture is in its own tab in construction. Nothing else is in it. Not caused here, just a note.

Maybe could move grid stuff to that category? I'd been meaning to, could tinker with it in a bit if desired.

**Pull request merged**